Src/Scripts/Include-GenerateUnitClassSourceCode.ps1

```diff
@@ -278,11 +280,30 @@ namespace UnitsNet
         public static $className Parse(string str, IFormatProvider formatProvider = null)
         {
             if (str == null) throw new ArgumentNullException("str");
-            string[] words = str.Split(new[] {" "}, StringSplitOptions.RemoveEmptyEntries);
-            if (words.Length < 2)
+
+            var numFormat = formatProvider != null ?
+                (NumberFormatInfo) formatProvider.GetFormat(typeof (NumberFormatInfo)) :
+                (NumberFormatInfo) CultureInfo.CurrentCulture.NumberFormat.Clone();
```

Comment:
CurrentInfo might be an improvement, I like it. Sort of makes it clear what the intention is.